automan/cluster_manager.py

```diff
@@ -368,6 +377,26 @@ def update(self, rebuild=True):
                 if rebuild:
                     self._rebuild(host, home)
 
+    def delete(self, sim_dir, remotes):
+        if remotes[-1] == 'all':
+            hosts = [h.get('host') for h in self.workers]
+            remotes = hosts.copy()
```

Comment:
Why is this needed?  The previous line has a copy of the hosts anyway?

Reply:
Since remote list is required later in the function. When 'all' is passed it copies worker list from the original host list.